go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go

```diff
@@ -189,39 +202,158 @@ func (itr *UnscopedDiffTableItr) calculateChangedTables(commit *doltdb.Commit) (
 		return nil, err
 	}
 
-	tablesMap := make(map[string]bool)
+	tableChanges := make([]tableChange, 0)
 	for _, delta := range deltas {
-		// Use toName by default for the table name by default, but if it's
-		// nil (e.g. when dropping a table), check to see if fromName is available.
-		tableName := delta.ToName
-		if len(tableName) == 0 {
-			if len(delta.FromName) > 0 {
-				tableName = delta.FromName
-			}
+		change, err := itr.processTableDelta(delta)
+		if err != nil {
+			return nil, err
 		}
 
-		if len(tableName) > 0 {
-			tablesMap[tableName] = true
-		}
+		tableChanges = append(tableChanges, *change)
 	}
 
 	// Not all commits mutate tables (e.g. empty commits)
-	if len(tablesMap) == 0 {
+	if len(tableChanges) == 0 {
 		return nil, nil
 	}
 
-	tables := make([]string, len(tablesMap))
-	i := 0
-	for key := range tablesMap {
-		tables[i] = key
-		i++
+	sort.Slice(tableChanges, func(i, j int) bool {
```

Comment:
Generally shouldn't do this unless the order is unstable. We should generally guarantee stable order for multiple executions of the same query but no particular defined order. Let the customer do an ORDER BY if they care about the order of results. If not this is wasted work.